tests/functional/tests/rgw/test_rgw.py

```diff
@@ -0,0 +1,44 @@
+import pytest
+
+
+class TestRGWs(object):
+
+    @pytest.mark.no_docker
+    def test_rgw_is_installed(self, node, host):
+        try:
+            result = host.package("radosgw").is_installed
+        except AssertionError:
+            result = host.package("ceph-radosgw").is_installed
+        assert result
+
+    def test_rgw_service_is_running(self, node, host):
+        service_name = "ceph-radosgw@rgw.{hostname}".format(
+            hostname=node["vars"]["inventory_hostname"]
+        )
+        assert host.service(service_name).is_running
+
+    def test_rgw_service_is_enabled(self, node, host):
+        service_name = "ceph-radosgw@rgw.{hostname}".format(
+            hostname=node["vars"]["inventory_hostname"]
+        )
+        assert host.service(service_name).is_enabled
+
+    @pytest.mark.no_docker
+    def test_rgw_is_up(self, node, host):
+        cmd = "sudo ceph --name client.bootstrap-rgw --keyring /var/lib/ceph/bootstrap-rgw/{cluster}.keyring --cluster={cluster} --connect-timeout 5 -f json -s | python -c 'import sys, json; print \"\n\".join([i for i in json.load(sys.stdin)[\"servicemap\"][\"services\"][\"rgw\"][\"daemons\"] if i == \"{hostname}\"])'".format(
+            hostname=node["vars"]["inventory_hostname"],
+            cluster=node["cluster_name"]
+        )
+        output = host.check_output(cmd)
+        phrase = "{hostname}".format(hostname=node["vars"]["inventory_hostname"])
+        assert phrase in output
+
+    @pytest.mark.docker
+    def test_rgw_is_up(self, node, host):
```

Comment:
`test_rgw_is_up()` is redefined here, it means it will never be tested on non-containerized deployment.